Lib/logging/__init__.py

```diff
@@ -2246,7 +2246,7 @@ def _showwarning(message, category, filename, lineno, file=None, line=None):
         logger = getLogger("py.warnings")
         if not logger.handlers:
             logger.addHandler(NullHandler())
-        logger.warning("%s", s)
+        logger.warning(str(s))
```

Comment:
You should add a comment here on why it works this way.

Reply:
will do

Reply:
Done ✅